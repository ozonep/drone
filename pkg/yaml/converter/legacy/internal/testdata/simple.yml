workspace:
  base: /go
  path: src/github.com/octocat/hello-world

pipeline:
  build:
    image: golang
    commands:
    - go get
    - go build
    volumes:
    - /tmp/go:/go/bin
    environment:
    - GOOS=linux
    - GOARCH=amd64

  test:
    image: golang:latest
    volumes:
    - /tmp/go:/go/bin
    commands:
    - go test -v
  
  docker:
    image: plugins/docker
    secrets:
    - docker_username
    - docker_password
    repo: octocat/hello-world
    when:
      branch: master

  slack:
    image: plugins/slack
    secrets:
    - source: token
      target: slack_token
    channel: general

services:
  database:
    image: mysql
    environment:
      MYSQL_USERNAME: foo
      MYSQL_PASSWORD: bar

branches:
- master