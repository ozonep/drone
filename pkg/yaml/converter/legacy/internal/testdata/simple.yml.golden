---
kind: pipeline
name: default

platform:
  os: linux
  arch: amd64

workspace:
  base: /go
  path: src/github.com/octocat/hello-world

steps:
- name: build
  pull: if-not-exists
  image: golang
  commands:
  - go get
  - go build
  environment:
    GOARCH: amd64
    GOOS: linux
  volumes:
  - name: 2f746d702f676f
    path: /go/bin

- name: test
  pull: if-not-exists
  image: golang:latest
  commands:
  - go test -v
  volumes:
  - name: 2f746d702f676f
    path: /go/bin

- name: docker
  pull: if-not-exists
  image: plugins/docker
  settings:
    repo: octocat/hello-world
  environment:
    DOCKER_PASSWORD:
      from_secret: docker_password
    DOCKER_USERNAME:
      from_secret: docker_username
  when:
    branch:
    - master

- name: slack
  pull: if-not-exists
  image: plugins/slack
  settings:
    channel: general
  environment:
    SLACK_TOKEN:
      from_secret: token

services:
- name: database
  pull: if-not-exists
  image: mysql
  environment:
    MYSQL_PASSWORD: bar
    MYSQL_USERNAME: foo

volumes:
- name: 2f746d702f676f
  host:
    path: /tmp/go

trigger:
  ref:
  - refs/pull/**
  - refs/pull-requests/**
  - refs/merge-requests/**
  - refs/heads/master

...
